src/Tools/ExternalAccess/Razor/DelegatingRazorSpanMappingService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    internal class DelegatingRazorSpanMappingService : ISpanMappingService
+    {
+        private readonly IRazorSpanMappingService _razorSpanMappingService;
+
+        public DelegatingRazorSpanMappingService(IRazorSpanMappingService razorSpanMappingService)
+        {
+            _razorSpanMappingService = razorSpanMappingService ?? throw new ArgumentNullException(nameof(razorSpanMappingService));
+        }
+
+        public async Task<ImmutableArray<MappedSpanResult>> MapSpansAsync(Document document, IEnumerable<TextSpan> spans, CancellationToken cancellationToken)
+        {
+            var razorSpans = await _razorSpanMappingService.MapSpansAsync(document, spans, cancellationToken).ConfigureAwait(false);
+            var roslynSpans = new MappedSpanResult[razorSpans.Length];
+            for (var i = 0; i < razorSpans.Length; i++)
+            {
+                var razorSpan = razorSpans[i];
+                roslynSpans[i] = new MappedSpanResult(razorSpan.FilePath, razorSpan.LinePositionSpan, razorSpan.Span);
+            }
+
+            return roslynSpans.ToImmutableArray();
```

Comment:
>roslynSpans [](start = 19, length = 11)

Can be simplified to `razorSpans.SelectAsArray(span => new MappedSpanResult(span.FilePath, span.LinePositionSpan, span.Span));` using [`SelectAsArray`](http://sourceroslyn.io/#Microsoft.CodeAnalysis/Collections/ArrayBuilderExtensions.cs,71) extension method.

Reply:
Eh, I like to avoid the terser statements like that because it's "less" code but harder to visibly consume/debug through.